<?xml version="1.0" encoding="UTF-8"?>
<assemblies>
  <assembly name="Mocha Tests" time="0.384" total="17" failed="12" passed="5" skipped="0" run-date="2021-07-18" run-time="17:29:04">
    <collection name="analyze(folderPath, callback)" total="10" failed="8" passed="2" time="0.002" skipped="0">
      <test name="analyze(folderPath, callback) should return an error in case a path to invalid folder was provided" time="0.001" result="passed">
      </test>
      <test name="analyze(folderPath, callback) should call callback function once" time="0" result="passed">
      </test>
      <test name="analyze(folderPath, callback) should pass null as an error in case of a success" time="0.001" result="failed">
        <failure exception-type="AssertionError">
          <message><![CDATA[expected [Error: Task not implemented] to equal null]]></message>
          <stack-trace><![CDATA[AssertionError: expected [Error: Task not implemented] to equal null
    at Context.<anonymous> (test\analyze.js:44:51)
    at processImmediate (internal/timers.js:439:21)]]></stack-trace>
        </failure>
      </test>
      <test name="analyze(folderPath, callback) should pass an object that has totalFiles number property" time="0" result="failed">
        <failure exception-type="AssertionError">
          <message><![CDATA[Target cannot be null or undefined.]]></message>
          <stack-trace><![CDATA[AssertionError: Target cannot be null or undefined.
    at Context.<anonymous> (test\analyze.js:48:52)
    at processImmediate (internal/timers.js:439:21)]]></stack-trace>
        </failure>
      </test>
      <test name="analyze(folderPath, callback) should pass an object that has totalSubFolders number property" time="0" result="failed">
        <failure exception-type="AssertionError">
          <message><![CDATA[Target cannot be null or undefined.]]></message>
          <stack-trace><![CDATA[AssertionError: Target cannot be null or undefined.
    at Context.<anonymous> (test\analyze.js:53:52)
    at processImmediate (internal/timers.js:439:21)]]></stack-trace>
        </failure>
      </test>
      <test name="analyze(folderPath, callback) should pass an object that has fileTypesInformation array property" time="0" result="failed">
        <failure exception-type="AssertionError">
          <message><![CDATA[Target cannot be null or undefined.]]></message>
          <stack-trace><![CDATA[AssertionError: Target cannot be null or undefined.
    at Context.<anonymous> (test\analyze.js:58:52)
    at processImmediate (internal/timers.js:439:21)]]></stack-trace>
        </failure>
      </test>
      <test name="analyze(folderPath, callback) should pass correct format of objects for fileTypesInformation array" time="0" result="failed">
        <failure exception-type="TypeError">
          <message><![CDATA[Cannot read property 'fileTypesInformation' of undefined]]></message>
          <stack-trace><![CDATA[TypeError: Cannot read property 'fileTypesInformation' of undefined
    at Context.<anonymous> (test\analyze.js:63:63)
    at processImmediate (internal/timers.js:439:21)]]></stack-trace>
        </failure>
      </test>
      <test name="analyze(folderPath, callback) should pass correct totalFiles value" time="0" result="failed">
        <failure exception-type="TypeError">
          <message><![CDATA[Cannot read property 'totalFiles' of undefined]]></message>
          <stack-trace><![CDATA[TypeError: Cannot read property 'totalFiles' of undefined
    at Context.<anonymous> (test\analyze.js:71:43)
    at processImmediate (internal/timers.js:439:21)]]></stack-trace>
        </failure>
      </test>
      <test name="analyze(folderPath, callback) should pass correct totalSubFolders value" time="0" result="failed">
        <failure exception-type="TypeError">
          <message><![CDATA[Cannot read property 'totalSubFolders' of undefined]]></message>
          <stack-trace><![CDATA[TypeError: Cannot read property 'totalSubFolders' of undefined
    at Context.<anonymous> (test\analyze.js:75:43)
    at processImmediate (internal/timers.js:439:21)]]></stack-trace>
        </failure>
      </test>
      <test name="analyze(folderPath, callback) should pass fileTypesInformation array that has correct information about extension" time="0" result="failed">
        <failure exception-type="TypeError">
          <message><![CDATA[Cannot read property 'fileTypesInformation' of undefined]]></message>
          <stack-trace><![CDATA[TypeError: Cannot read property 'fileTypesInformation' of undefined
    at Context.<anonymous> (test\analyze.js:79:43)
    at processImmediate (internal/timers.js:439:21)]]></stack-trace>
        </failure>
      </test>
    </collection>
    <collection name="CLI" total="7" failed="4" passed="3" time="0.382" skipped="0">
      <test name="CLI It should exit with non-zero code (Throw an error) if passed incorrect command name" time="0.095" result="passed">
      </test>
      <test name="CLI It should exit with non-zero code (Throw an error) if passed incorrect arguments for rimraf command" time="0.093" result="passed">
      </test>
      <test name="CLI It should exit with non-zero code (Throw an error) if passed incorrect arguments for rename command" time="0.096" result="passed">
      </test>
      <test name="CLI It should delete a folder with all it content when using rimraf command." time="0.001" result="failed">
        <failure exception-type="Error">
          <message><![CDATA[EEXIST: file already exists, mkdir 'C:\Users\Svitlana_Dasgupta\Desktop\Projects\Learning\JS Basic Ecosystem\HW\HW1\HW_done\part1\testDir']]></message>
          <stack-trace><![CDATA[Error: EEXIST: file already exists, mkdir 'C:\Users\Svitlana_Dasgupta\Desktop\Projects\Learning\JS Basic Ecosystem\HW\HW1\HW_done\part1\testDir']]></stack-trace>
        </failure>
      </test>
      <test name="CLI It should delete a file when using rimraf command." time="0.049" result="failed">
        <failure exception-type="AssertionError">
          <message><![CDATA[expected 1 to equal 0]]></message>
          <stack-trace><![CDATA[AssertionError: expected 1 to equal 0
    at Context.<anonymous> (test\index.js:85:25)
    at processTicksAndRejections (internal/process/task_queues.js:93:5)]]></stack-trace>
        </failure>
      </test>
      <test name="CLI It should rename a file when using rename command." time="0.048" result="failed">
        <failure exception-type="AssertionError">
          <message><![CDATA[expected 1 to equal 0]]></message>
          <stack-trace><![CDATA[AssertionError: expected 1 to equal 0
    at Context.<anonymous> (test\index.js:101:25)
    at processTicksAndRejections (internal/process/task_queues.js:93:5)]]></stack-trace>
        </failure>
      </test>
      <test name="CLI It should rename a folder when using rename command." time="0" result="failed">
        <failure exception-type="Error">
          <message><![CDATA[EEXIST: file already exists, mkdir 'C:\Users\Svitlana_Dasgupta\Desktop\Projects\Learning\JS Basic Ecosystem\HW\HW1\HW_done\part1\folder']]></message>
          <stack-trace><![CDATA[Error: EEXIST: file already exists, mkdir 'C:\Users\Svitlana_Dasgupta\Desktop\Projects\Learning\JS Basic Ecosystem\HW\HW1\HW_done\part1\folder']]></stack-trace>
        </failure>
      </test>
    </collection>
  </assembly>
</assemblies>